crates/subspace-farmer/src/utils/archival_storage_pieces.rs

```diff
@@ -36,3 +36,15 @@ impl ArchivalStoragePieces {
         Ok(())
     }
 }
+
+impl PeerInfoProvider for ArchivalStoragePieces {
+    fn peer_info(&self) -> PeerInfo {
+        let exported_filter = self.cuckoo_filter.lock().export();
+        let data = serde_scale::to_vec(&exported_filter).expect("Serialization always works.");
```

Comment:
[ExportedCuckooFilter](https://docs.rs/cuckoofilter/latest/cuckoofilter/struct.ExportedCuckooFilter.html) has two public fields, for which you can create a mirror data structure with anything you want derived on it. I don't think it is a better idea to go through `serde_scale` instead.

Also I'm not sure having `data` field as `Vec<u8>` is ideal, I'd rather have the whole `PeerInfo` as an enum, then you'd have `PeerInfo::Farmer {cuckoo_filter: CuckooFilterExported}`, which is nicely serialized/deserialized for you.

And `Debug` will be trivially derivable.

Reply:
We use SCALE within the project but most types provide `serde` features. `serde_scale` is a good option IMHO. I don't see downsides except for an additional dependency and we avoid new serialization code. 

`PeerInfo::Farmer {cuckoo_filter: CuckooFilterExported}` - this was my original plan but I'm not sure the type will stay the same in the end. This is why I created a more abstract struct and added TODO both in the code and in the PR description for this case. If the requirements don't change I will change it to the proposed enum. I don't like auto-derived Debug for this type because it prints the whole filter. I added a custom implementation to show only a vector length.

Reply:
> We use SCALE within the project but most types provide serde features. serde_scale is a good option IMHO. I don't see downsides except for an additional dependency and we avoid new serialization code.

I'm not exactly sure how `serde_scale` works, but it is surely less efficient than direct implementation. I see no reason to use it if we can have proper and efficient implementation instead. Feels like absolutely unnecessary workaround.

> PeerInfo::Farmer {cuckoo_filter: CuckooFilterExported} - this was my original plan but I'm not sure the type will stay the same in the end. This is why I created a more abstract struct and added TODO both in the code and in the PR description for this case. If the requirements don't change I will change it to the proposed enum.

All of such decisions explicitly do not support upgrades. Why don't you do `PeerInfo::V0(PeerInfoV0)` then with `PeerInfoV0::Farmer {cuckoo_filter: CuckooFilterExported}` so you can always introduce V1 if that is a concern?

I really dislike when we have two field, where one is an enum explaining what the other field contains. You would do that in C because C doesn't support the kinds of enums Rust does, but here it feels like completely unnecessary awkwardness for no benefit.